### ISSUE TITLE
在aarch64服务器上编译安装paddlepaddle-gpu版本，系统显示安装成功，但是利用paddleClas，paddleNLP的GPU后端训练所有模型，loss基本为nan。

### ISSUE THREAD

**Issue by yefeichen1** [PaddlePaddle/Paddle#47839]
### bug描述 Describe the Bug

1. 报错信息
1.1 系统显示安装成功，但是执行官方提供的profile案例ernie时执行结果与预期不符
python3
import paddle
paddle.utils.run_check()
系统显示安装成功
oot@a1add718b0a0:/workspace/paddle/PaddleNLP_aarch64# python3
Python 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:56:26)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import paddle
>>> paddle.utils.run_check()
Running verify PaddlePaddle program ...
I1110 07:09:07.133450 75773 interpretercore.cc:243] New Executor is Running.
W1110 07:09:07.133654 75773 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 07:09:07.145354 75773 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
I1110 07:09:08.895836 75773 interpreter_util.cc:534] Standalone Executor is Used.
PaddlePaddle works well on 1 GPU.
PaddlePaddle works well on 1 GPUs.
PaddlePaddle is installed successfully! Let's start deep learning with PaddlePaddle now.

但是运行profile中的python ernie.py--model_dirernieL3H128_model/，结果输出[0,0.]与预期的结果，[正确输出应为：[0.49879393, 0.5012061 ]不同.

1.2 利用paddleClas和paddleNLP的GPU后端训练resnet50模型和bert模型，loss均为nan.
paddleClas报错
[2022/11/10 06:42:03] ppcls INFO: train with paddle 0.0.0 and device Place(gpu:0)
W1110 06:42:33.366624 11392 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 06:42:33.380874 11392 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
[2022/11/10 06:42:34] ppcls WARNING: The training strategy provided by PaddleClas is based on 4 gpus. But the number of gpu is 1 in current training. Please modify the stategy (learning rate, batch size and so on) if use this config to train.
[2022/11/10 06:42:40] ppcls INFO: [Train][Epoch 1/120][Iter: 0/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 0.00000, loss: 0.00000, batch_cost: 5.84739s, reader_cost: 1.18457, ips: 10.94505 samples/s, eta: 162 days, 13:57:54
[2022/11/10 06:42:44] ppcls INFO: [Train][Epoch 1/120][Iter: 10/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 86.24393, loss: 86.24393, batch_cost: 0.40814s, reader_cost: 0.12773, ips: 156.81021 samples/s, eta: 11 days, 8:20:54
[2022/11/10 06:42:50] ppcls INFO: [Train][Epoch 1/120][Iter: 20/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 60.23386, loss: 60.23386, batch_cost: 0.56701s, reader_cost: 0.29120, ips: 112.87336 samples/s, eta: 15 days, 18:21:38
[2022/11/10 06:42:57] ppcls INFO: [Train][Epoch 1/120][Iter: 30/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 51.00448, loss: 51.00448, batch_cost: 0.61040s, reader_cost: 0.33597, ips: 104.84937 samples/s, eta: 16 days, 23:18:52
[2022/11/10 06:43:04] ppcls INFO: [Train][Epoch 1/120][Iter: 40/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.62121s, reader_cost: 0.34818, ips: 103.02555 samples/s, eta: 17 days, 6:31:23
[2022/11/10 06:43:10] ppcls INFO: [Train][Epoch 1/120][Iter: 50/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63159s, reader_cost: 0.35759, ips: 101.33211 samples/s, eta: 17 days, 13:26:56
[2022/11/10 06:43:17] ppcls INFO: [Train][Epoch 1/120][Iter: 60/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63364s, reader_cost: 0.35921, ips: 101.00355 samples/s, eta: 17 days, 14:49:05
[2022/11/10 06:43:23] ppcls INFO: [Train][Epoch 1/120][Iter: 70/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63834s, reader_cost: 0.36346, ips: 100.25998 samples/s, eta: 17 days, 17:57:07
[2022/11/10 06:43:30] ppcls INFO: [Train][Epoch 1/120][Iter: 80/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.64019s, reader_cost: 0.36499, ips: 99.97039 samples/s, eta: 17 days, 19:11:03
[2022/11/10 06:43:36] ppcls INFO: [Train][Epoch 1/120][Iter: 90/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.64001s, reader_cost: 0.36429, ips: 99.99864 samples/s, eta: 17 days, 19:03:42

paddleNLP的bert模型
I1110 07:01:30.791853 75344 interpretercore.cc:243] New Executor is Running.
W1110 07:01:30.821660 75344 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 07:01:30.832885 75344 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
I1110 07:01:57.526959 75344 interpreter_util.cc:534] Standalone Executor is Used.
2022-11-10 07:01:59,182-INFO: step_idx: 0, epoch: 0, batch: 0, avg loss: nan, normalized loss: nan, ppl: nan
2022-11-10 07:02:40,019-INFO: step_idx: 100, epoch: 0, batch: 100, avg loss: -0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.45 step/s, batch_cost: 0.40806 sec, reader_cost: 0.00030 sec, tokens: -159587898190006346237561798656, ips: -3910920245577268960866336768.00000 words/sec
2022-11-10 07:03:20,611-INFO: step_idx: 200, epoch: 0, batch: 200, avg loss: 0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.47 step/s, batch_cost: 0.40561 sec, reader_cost: 0.00028 sec, tokens: -160143334344150069733652168704, ips: -3948257979291911143561363456.00000 words/sec
2022-11-10 07:04:01,281-INFO: step_idx: 300, epoch: 0, batch: 300, avg loss: 0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.46 step/s, batch_cost: 0.40639 sec, reader_cost: 0.00028 sec, tokens: -320879396286813222064931995648, ips: -7895805655588062461626941440.00000 words/sec
2022-11-10 07:04:41,698-INFO: step_idx: 400, epoch: 0, batch: 400, avg loss: 0.895865, normalized loss: -0.485088, ppl: 2.449454, avg_speed: 2.48 step/s, batch_cost: 0.40385 sec, reader_cost: 0.00028 sec, tokens: -504399071057308146759276429312, ips: -12489683768828281202315100160.00000 words/sec

### 其他补充信息 Additional Supplementary Information

1. 系统环境
GPU型号：A100
CUDA版本号：CUDA 11.6
操作系统：ky10 aarch64

docker镜像: nvcr.io/nvidia/pytorch:22.04-py3
docker内部的环境：
gcc 9.4.0
g++ 9.4.0
patchelf 0.14.5
python3..8.13

2.  复现步骤
编译参考链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/install/compile/arm-compile.html
其中，修改cmake的编译选项为：
cmake .. -DPY_VERSION=3.8 -DWITH_ARM=ON -DWITH_AARCH64=ON  -DWITH_GPU=ON -DWITH_PSCORE=OFF -DON_INFER=ON -DWITH_TESTING=OFF -DWITH_XBYAK=OFF -DCMAKE_CXX_FLAGS="-Wno-error -w"

3. 报错信息
3.1 系统显示安装成功，但是执行官方提供的profile案例ernie时执行结果与预期不符
python3
import paddle
paddle.utils.run_check()
系统显示安装成功
oot@a1add718b0a0:/workspace/paddle/PaddleNLP_aarch64# python3
Python 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:56:26)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import paddle
>>> paddle.utils.run_check()
Running verify PaddlePaddle program ...
I1110 07:09:07.133450 75773 interpretercore.cc:243] New Executor is Running.
W1110 07:09:07.133654 75773 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 07:09:07.145354 75773 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
I1110 07:09:08.895836 75773 interpreter_util.cc:534] Standalone Executor is Used.
PaddlePaddle works well on 1 GPU.
PaddlePaddle works well on 1 GPUs.
PaddlePaddle is installed successfully! Let's start deep learning with PaddlePaddle now.

但是运行profile中的python ernie.py--model_dirernieL3H128_model/，结果输出[0,0.]与预期的结果
[正确输出应为：[0.49879393, 0.5012061 ]不同.

3.2 利用paddleClas和paddleNLP的GPU后端训练resnet50模型和bert模型，loss均为nan.
paddleClas报错
[2022/11/10 06:42:03] ppcls INFO: train with paddle 0.0.0 and device Place(gpu:0)
W1110 06:42:33.366624 11392 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 06:42:33.380874 11392 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
[2022/11/10 06:42:34] ppcls WARNING: The training strategy provided by PaddleClas is based on 4 gpus. But the number of gpu is 1 in current training. Please modify the stategy (learning rate, batch size and so on) if use this config to train.
[2022/11/10 06:42:40] ppcls INFO: [Train][Epoch 1/120][Iter: 0/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 0.00000, loss: 0.00000, batch_cost: 5.84739s, reader_cost: 1.18457, ips: 10.94505 samples/s, eta: 162 days, 13:57:54
[2022/11/10 06:42:44] ppcls INFO: [Train][Epoch 1/120][Iter: 10/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 86.24393, loss: 86.24393, batch_cost: 0.40814s, reader_cost: 0.12773, ips: 156.81021 samples/s, eta: 11 days, 8:20:54
[2022/11/10 06:42:50] ppcls INFO: [Train][Epoch 1/120][Iter: 20/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 60.23386, loss: 60.23386, batch_cost: 0.56701s, reader_cost: 0.29120, ips: 112.87336 samples/s, eta: 15 days, 18:21:38
[2022/11/10 06:42:57] ppcls INFO: [Train][Epoch 1/120][Iter: 30/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: 51.00448, loss: 51.00448, batch_cost: 0.61040s, reader_cost: 0.33597, ips: 104.84937 samples/s, eta: 16 days, 23:18:52
[2022/11/10 06:43:04] ppcls INFO: [Train][Epoch 1/120][Iter: 40/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.62121s, reader_cost: 0.34818, ips: 103.02555 samples/s, eta: 17 days, 6:31:23
[2022/11/10 06:43:10] ppcls INFO: [Train][Epoch 1/120][Iter: 50/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63159s, reader_cost: 0.35759, ips: 101.33211 samples/s, eta: 17 days, 13:26:56
[2022/11/10 06:43:17] ppcls INFO: [Train][Epoch 1/120][Iter: 60/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63364s, reader_cost: 0.35921, ips: 101.00355 samples/s, eta: 17 days, 14:49:05
[2022/11/10 06:43:23] ppcls INFO: [Train][Epoch 1/120][Iter: 70/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.63834s, reader_cost: 0.36346, ips: 100.25998 samples/s, eta: 17 days, 17:57:07
[2022/11/10 06:43:30] ppcls INFO: [Train][Epoch 1/120][Iter: 80/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.64019s, reader_cost: 0.36499, ips: 99.97039 samples/s, eta: 17 days, 19:11:03
[2022/11/10 06:43:36] ppcls INFO: [Train][Epoch 1/120][Iter: 90/20019]lr(PiecewiseDecay): 0.10000000, top1: 0.00000, top5: 0.00000, CELoss: nan, loss: nan, batch_cost: 0.64001s, reader_cost: 0.36429, ips: 99.99864 samples/s, eta: 17 days, 19:03:42

paddleNLP的bert模型
I1110 07:01:30.791853 75344 interpretercore.cc:243] New Executor is Running.
W1110 07:01:30.821660 75344 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 8.0, Driver API Version: 11.6, Runtime API Version: 11.6
W1110 07:01:30.832885 75344 gpu_resources.cc:91] device: 0, cuDNN Version: 8.4.
I1110 07:01:57.526959 75344 interpreter_util.cc:534] Standalone Executor is Used.
2022-11-10 07:01:59,182-INFO: step_idx: 0, epoch: 0, batch: 0, avg loss: nan, normalized loss: nan, ppl: nan
2022-11-10 07:02:40,019-INFO: step_idx: 100, epoch: 0, batch: 100, avg loss: -0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.45 step/s, batch_cost: 0.40806 sec, reader_cost: 0.00030 sec, tokens: -159587898190006346237561798656, ips: -3910920245577268960866336768.00000 words/sec
2022-11-10 07:03:20,611-INFO: step_idx: 200, epoch: 0, batch: 200, avg loss: 0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.47 step/s, batch_cost: 0.40561 sec, reader_cost: 0.00028 sec, tokens: -160143334344150069733652168704, ips: -3948257979291911143561363456.00000 words/sec
2022-11-10 07:04:01,281-INFO: step_idx: 300, epoch: 0, batch: 300, avg loss: 0.000000, normalized loss: -1.380953, ppl: 1.000000, avg_speed: 2.46 step/s, batch_cost: 0.40639 sec, reader_cost: 0.00028 sec, tokens: -320879396286813222064931995648, ips: -7895805655588062461626941440.00000 words/sec
2022-11-10 07:04:41,698-INFO: step_idx: 400, epoch: 0, batch: 400, avg loss: 0.895865, normalized loss: -0.485088, ppl: 2.449454, avg_speed: 2.48 step/s, batch_cost: 0.40385 sec, reader_cost: 0.00028 sec, tokens: -504399071057308146759276429312, ips: -12489683768828281202315100160.00000 words/sec

**Comment by wanghuancoder** [comment]
感谢您的问题反馈！目前Paddle还没有在这个环境下进行过测试，也没有发这种环境的包。我将反馈给相关同学进一步跟进
GPU型号：A100
CUDA版本号：CUDA 11.6
操作系统：ky10 aarch64

**Comment by wanghuancoder** [comment]
@pangyoki @qili93 
我们应该有A100的测试，NPU上应该是用过ARM。但ARM+A100应该没有使用过。请两位看看这个问题有什么解决办法吗？

**Comment by yefeichen1** [comment]
也不只针对A100的GPU，T4也有这种情况。谢谢




1040248883
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/Paddle"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月10日(星期四) 下午4:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/Paddle] 在aarch64服务器上编译安装paddlepaddle-gpu版本，系统显示安装成功，但是利用paddleClas，paddleNLP的GPU后端训练所有模型，loss基本为nan。 (Issue #47839)





 
@pangyoki @qili93
 我们应该有A100的测试，NPU上应该是用过ARM。但ARM+A100应该没有使用过。请两位看看这个问题有什么解决办法吗？

**Comment by wanghuancoder** [comment]
@pangyoki 咱们T4有测试过或者发包吗？
